chalk-solve/src/infer/invert.rs

```diff
@@ -150,7 +150,7 @@ where
             .inverted_ty
             .entry(universe)
             .or_insert_with(|| table.new_variable(universe.ui))
-            .to_ty(self.interner())
+            .to_ty(self.interner(), TyKind::General)
```

Comment:
maybe make `to_ty` assume `TyKind::General`

Reply:
and add `to_ty_with_kind` or something?

Reply:
More often than not (except for the tests), the kind is available because `to_ty` is called as part of a transformation on the inference variable. Maybe instead I can add a `to_new_ty(interner)` method to signify that this will return a new, general ty variable?

Reply:
Hm, looking at it again, the kind is only available once. Maybe some refactoring removed the other usages. I'll make `to_ty` assume `TyKind::General`.